src/app/clusters/service-area-server/service-area-server.h

```diff
@@ -0,0 +1,751 @@
+     * @param aInstance A pointer to the Instance object related to this delegate object.
+     */
+    void SetInstance(Instance * aInstance) { mInstance = aInstance; }
+
+
+    //*************************************************************************
+    // volatile attribute support
+
+    /**
+     * @brief give the device a chance to update current location
+     * @param prevCurrentLocation "current" current location value - used to determine if update notification is triggered
+     * @note this is called by the instance function GetCurrentLocation() to ensure the latest value is available
+     * @note default implementation can be used if current location is non-volatile (always expected to be up-to-date)
+    */
+    virtual void HandleVolatileCurrentLocation(const DataModel::Nullable<uint32_t> prevCurrentLocation) = 0;
+
+    /**
+     * @brief give the device a chance to update estimated end time
+     * @param prevEstimatedEndTime "current" estimated end time value - used to determine if update notification is triggered
+     * @note this is called by the instance function GetEstimatedEndTime() to ensure the latest value is available
+     * @note default implementation can be used if progress entries are non-volatile (always expected to be up-to-date)
+    */
+    virtual void HandleVolatileEstimatedEndTime(const DataModel::Nullable<uint32_t> prevEstimatedEndTime) = 0;
+
+    /**
+     * @brief give the device a chance to update progress elements
+     * @note this is called by the instance functions ReadProgress() and  GetEstimatedEndTime() to ensure the latest values are available
+     * @note default implementation can be used if progress elements are non-volatile (always expected to be up-to-date)
+     * @note The EstimatedTime field is the most likely to be a volatile value
+    */
+    virtual void HandleVolatileProgressList() = 0;
+
+
+    //*************************************************************************
+    // Selected Locations "pruning" support
+
+    /**
+     * @brief the device has changed supported maps or supported locations, leading to a change in selected locations.
+     * @note If the device does not make or allow changes to supported maps or supported locations while operating, it's likely no action is needed
+     * @note If selected locations is changed such that no locations are selected and the device is operating, the device must stop.
+     * @note If selected locations is changed and the device is operating, the device may need to update it's operating sequence
+     * @note If the device is operating and does not allow changing selected locations while operating, the device still needs to handle this change somehow (e.g. abort operation).
+     *  (changes to maps or supported locations may result in changes in selected locations, so change may occur as a side-effect of updates)
+     */
+    virtual void HandleSelectedLocationsChanged() = 0;
```

Comment:
The requirement changed and complicated things.

Reply:
This has been removed. Not sure when.